### PR TITLE
[MRG] Fix: Masking in NiftiLabelsMasker doesn't work

### LINKED ISSUE #2233: Masking in NiftiLabelsMasker doesn't work.
I want to use `NiftiLabelsMasker` to extract voxels of interest using the Destrieux-Atlas. I am working with MRI-Data (not fMRI) so the 4th dimension represents different subjects, not different time points for a single subject. I created a dummy script and tried to use `NiftiLabelsMasker` on the OASIS dataset (which is available through nilearn.datasets.fetch_oasis_vbm). I also played with the mask image (I tried to create a mask that reduces the atlas image to regions 1 and 2). But it seems that the masking doesn't work: Although I provide `NiftiLabelsMasker` with destrieux_atlas_roi_mask I get 148 data columns (which seems to be all 148 regions reported in this paper). I expect NiftiLabelsMasker to output only 2 columns. Here's my code example:

```
from nilearn.image import load_img
from nilearn.datasets import fetch_atlas_destrieux_2009
from nilearn.datasets import fetch_oasis_vbm
from nilearn.image import math_img

from nilearn.input_data import NiftiLabelsMasker

# niftilabels masker cache
niftilabelsmasker_cache = './niftilabelsmasker_cache/'

# fetch OASIS dataset
oasis_img_paths = fetch_oasis_vbm(n_subjects=10)['gray_matter_maps']

# load destrieux atlas
destrieux_atlas_dict = fetch_atlas_destrieux_2009()
destrieux_atlas = load_img(fetch_atlas_destrieux_2009()['maps'])

# get only region 1 and 2
destrieux_atlas_roi_mask = math_img('np.where(((img == 1) | (img == 2)),1,0)',img=destrieux_atlas)

# extract data
niftilabelsmasker = NiftiLabelsMasker(labels_img=destrieux_atlas,
                                      mask_img=destrieux_atlas_roi_mask,
                                      memory=niftilabelsmasker_cache)

img_data = niftilabelsmasker.fit_transform(oasis_img_paths)
```

This issue might be related to #1056 and #1257.

Pinging @bthirion here to confirm that I posted this as issue on GitHub. I originally asked [this question on neurostars](https://neurostars.org/t/problems-with-using-niftilabelsmasker/5483/2).

### PR DESCRIPTION
**Pull request by jeromedockes**
fixes #2233

### CODE REVIEW

**Comment by emdupre** [comment]
Wanted to check in on this -- do we just need to resolve the merge conflicts, or were you waiting on another review @jeromedockes ?

**Comment by jeromedockes** [comment]
> Wanted to check in on this -- do we just need to resolve the merge conflicts, or were you waiting on another review @jeromedockes ?

I had sort of forgotten about it ... I will fix conflicts and let you know

**Comment by emdupre** [comment]
I'll merge this later today unless I hear otherwise from you @jeromedockes (or if you want to merge it first, go ahead !).

**Comment by jeromedockes** [comment]
> I'll merge this later today unless I hear otherwise from you @jeromedockes (or if you want to merge it first, go ahead !).

yes, I think you can go ahead and merge
